contracts/src/LightClientV2.sol

```diff
@@ -0,0 +1,208 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.0;
+
+import { BN254 } from "bn254/BN254.sol";
+import { LightClient } from "./LightClient.sol";
+import { IPlonkVerifier } from "./interfaces/IPlonkVerifier.sol";
+import { PlonkVerifierV2 as PV } from "./libraries/PlonkVerifierV2.sol";
+import { LightClientStateUpdateVKV2 as VkLib } from "./libraries/LightClientStateUpdateVKV2.sol";
+
+/// @notice LightClient V2: with stake table snapshot update during epoch change.
+contract LightClientV2 is LightClient {
+    /// @notice When entering a new epoch and a new stake table snapshot.
+    event NewEpoch(uint64 epoch);
+
+    /// @notice PlonkVerifier contract address
+    address public _verifier;
```

Comment:
@alxiong can you remind why we are making the verifier a contract and not an external library?

Reply:
just to reduce contract size. we don't inline the whole `verify()` logic into `LightClientV2`. 
Is this okay? do you still prefer the old way?

Reply:
External libraries are also separately deployed, and linked by injecting the address to jump to in the caller contract bytecode. So the argument about contract size IMO does not apply. I think this is a change without any benefit and therefore worse than keeping it the same but let me know if you disagree.

Reply:
okay, even if the contract size argument doesn't apply, avoiding "injecting the linked address in caller's bytecode" is a minor plus. 
as doing so requires:

https://github.com/EspressoSystems/espresso-network/blob/ccd9e6246e640c32b753e3e4b7c91fde4b591123/utils/src/deployer.rs#L217-L227

okay, so are we reverting them back to:

- `PlonkVerifierV2` as `library`
- linked `PV2` to `LightClientV2`?

Reply:
done in [1350b70](https://github.com/EspressoSystems/espresso-network/pull/2706/commits/1350b70696c6f3625d5859897e7217dce8d08c94)